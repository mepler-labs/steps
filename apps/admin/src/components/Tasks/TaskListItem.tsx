import { Box, Flex } from 'grid-styled';
import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { blue, green, lightGrey, white } from 'styles/colors';
import { remCalc, sansSerif } from 'styles/type';

const StyledLink = styled(Link)`
  color: ${green};
  font-size: 0.8em;
  margin-top: 0.5em;
  text-decoration: none;
  text-transform: uppercase;
`;

export default ({ setTaskStatus, task, url, index }) => {
  const toggleTaskStatus = e => {
    const status = task.status === 'COMPLETED' ? 'ACTIVE' : 'COMPLETED';
    setTaskStatus(task, status);
  };

  const checked =
    task.status === 'COMPLETED' ? 'check_circle' : 'check_circle_outline';

  return (
    <Background
      flexDirection="column"
      alignItems="flex-start"
      justifyContent="center"
      p={20}
      width={1}
    >
      <Flex flexDirection="row" alignItems="center">
        <div onClick={toggleTaskStatus}>
          <i className={`material-icons ${checked}`}>{checked}</i>
        </div>
        <Title>{task.title}</Title>
      </Flex>
      <Box ml={34}>
        <StyledLink to={`${url}/${task.id}`}>View Steps &rarr;</StyledLink>
      </Box>
    </Background>
  );
};

const Background = styled(Flex)`
  background: ${white};
  border-bottom-right-radius: 4px;
  border-top-right-radius: 4px;

  i.material-icons {
    cursor: pointer;
    width: 24px;
  }

  .check_circle {
    color: ${blue};
  }

  .check_circle_outline {
    color: ${lightGrey};
  }
`;

const Title = styled.h3`
  font-family: ${sansSerif};
  font-size: ${remCalc(24)};
  font-weight: 500;
  margin-top: 0;
  margin-left: 10px;
  margin-right: 0;
  margin-bottom: ${remCalc(10)};
`;
