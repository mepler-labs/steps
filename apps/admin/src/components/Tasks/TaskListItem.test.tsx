import { shallow } from 'enzyme';
import React from 'react';

import TaskListItem from './TaskListItem';

describe('TaskListItem.tsx', () => {
  it('renders correctly', () => {
    const task = {
      title: 'My task',
      description: 'task description',
      status: 'ACTIVE',
    };
    const wrapper = shallow(
      <TaskListItem
        key="key"
        setTaskStatus={() => {}}
        task={task}
        url="foo/bar"
        index={1}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can toggle the task status', () => {
    const setTaskStatus = jest.fn();
    const task = {
      title: 'My task',
      description: 'task description',
      status: 'ACTIVE',
    };
    const wrapper = shallow(
      <TaskListItem
        key="key"
        setTaskStatus={setTaskStatus}
        task={task}
        url="foo/bar"
        index={1}
      />,
    );

    const toggler = wrapper.find('.task-completed');

    toggler.simulate('click');

    expect(setTaskStatus).toHaveBeenCalledWith(task, 'COMPLETED');
  });
});
