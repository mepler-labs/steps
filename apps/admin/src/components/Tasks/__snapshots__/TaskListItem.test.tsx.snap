// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem.tsx renders correctly 1`] = `
<Styled(Flex)
  alignItems="flex-start"
  flexDirection="column"
  justifyContent="center"
  p={20}
  width={1}
>
  <Flex
    alignItems="center"
    flexDirection="row"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <div
      className="task-completed"
      onClick={[Function]}
    >
      <i
        className="material-icons check_circle_outline"
      >
        check_circle_outline
      </i>
    </div>
    <styled.h3>
      My task
    </styled.h3>
  </Flex>
  <Box
    ml={34}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Styled(Link)
      to="foo/bar/undefined"
    >
      View Steps â†’
    </Styled(Link)>
  </Box>
</Styled(Flex)>
`;
